src/deployment/cd.md

```diff
@@ -193,6 +193,152 @@ secrets in pull requests that you accept and merge.
+testing, and distributing apps and frameworks for Apple platforms.
+
+### Requirements
+
+* Xcode Version 13.4.1 or higher.
+* Be enrolled in the [Apple Developer Program][].
+
+### Custom build script
+
+Xcode Cloud recognizes three different [custom build scripts][] that can be
+used to perform additional tasks at a designated time.
+
+{{site.alert.note}}
+  The temporary build environment that Xcode Cloud uses includes tools that are
+  part of macOS and Xcode&mdash;for example, Python&mdash;and additionally Homebrew to
+  support installing third-party dependencies and tools.
+{{site.alert.end}}
+
+#### Post-clone script
+
+Leverage the `post-clone` script that runs after Xcode Cloud
+clones your Git repository using the following instructions:
+
+1. Navigate to the `ios` folder in your project root.
+
+2. Create a new folder named `ci_scripts`. 
+
+3. Create a file named `ci_post_clone.sh` in the `ci_scripts` folder.
+    * Add a shebang line at the top of the script.
+      ```sh
+      #!/bin/sh
+      ```
+
+4. This file should be added to your git repository and marked as executable.
+    ```terminal
+    $ git add --chmod=+x ios/ci_scripts/ci_post_clone.sh
+    ```
+
+#### Working directory
+
+Xcode Cloud includes a set of [predefined environment variables][], such as
+`$CI_WORKSPACE`, which is the location of your cloned repository.
+
+The working directory for Xcode Cloud's custom build scripts is the `ci_scripts/`
+folder, hence you need to navigate to the root of your cloned project by adding
+the following command to your `ci_post_clone` script below the shebang line.
+
+```sh
+cd $CI_WORKSPACE
+```
+
+#### Install Flutter
+
+The Flutter SDK must be installed manually, [Download Flutter via Git][] by
+adding the commands below to your `ci_post_clone` script:
+
+```sh
+git clone https://github.com/flutter/flutter.git -b stable $HOME/flutter
+export PATH="$PATH:$HOME/flutter/bin"
+```
+
+#### Install CocoaPods
+
+CocoaPods can be installed by adding the following to your `ci_post_clone` script:
+
+```sh
+HOMEBREW_NO_AUTO_UPDATE=1 # disable homebrew's automatic updates
+brew install cocoapods
+```
+
+#### Complete script
+
+```sh
+#!/bin/sh
+
+# by default, the execution directory of this script is the ci_scripts directory.
+# CI_WORKSPACE is the directory of your cloned repo.
+cd $CI_WORKSPACE # change working directory to the root of your cloned repo.
+
+# Install Flutter using git.
+git clone https://github.com/flutter/flutter.git -b stable $HOME/flutter
+export PATH="$PATH:$HOME/flutter/bin"
+
+# Install Flutter artifacts for iOS (--ios), or macOS (--macos) platforms.
+flutter precache --ios
+
+# Install Flutter dependencies.
+flutter pub get
+
+# Install CocoaPods using Homebrew.
+HOMEBREW_NO_AUTO_UPDATE=1 # disable homebrew's automatic updates.
+brew install cocoapods
+
+# Install CocoaPods dependencies".
+cd ios && pod install # run `pod install` in the `ios` directory.
```

Comment:
@jmagman  Ahh, I wasn't aware of the `--config-only` flag and this is exactly what it's intended for. 

> Although this isn't an available flag for macOS yet.

Though I do agree that it might be worth leaving as is to maintain consistency across iOS/macOS.